packages/rocketchat-lib/client/MessageAction.js

```diff
@@ -103,7 +103,12 @@ Meteor.startup(function() {
 			const message = this._arguments[1];
 			const input = instance.find('.input-message');
 			const url = RocketChat.MessageAction.getPermaLink(message._id);
-			const text = `[ ](${ url }) @${ message.u.username } `;
+			let text = `[ ](${ url }) `;
+
+			if (Session.get('openedRoom').indexOf(Meteor.userId()) === -1) {
```

Comment:
I think it's safe doing `RocketChat.models.Rooms.findOne(message.rid, { fields: { t: 1 } })` to get room's type =)